### PR TITLE
Port reference highlighting tests

### PR DESCRIPTION
**Pull request by rchande**
tag @dotnet/roslyn-ide for review

### CODE REVIEW

**Comment by rchande** [comment]
Replaces:
https://github.com/dotnet/roslyn-internal/blob/master/Closed/Hosting/RoslynTaoActions/IntegrationTests/BasicReferenceHighlighting.xml
https://github.com/dotnet/roslyn-internal/blob/master/Closed/Hosting/RoslynTaoActions/IntegrationTests/CSharpReferenceHighlighting.xml

File: src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReferenceHighlighting.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public void Highlighting()
+        {
+            var markup = @"
+class {|definition:C|}
+{
+    void M<T>({|reference:C|} c) where T : {|reference:C|}
+    {
+        {|reference:C|} c = new {|reference:C|}();
```
Comment (Pilchie) [inline src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReferenceHighlighting.cs]: Would be nice to very the written reference versus reference stuff.

File: src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReferenceHighlighting.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            VisualStudio.Editor.PlaceCaret(marker, charsOffset: -1);
+            VisualStudio.Workspace.WaitForAsyncOperations(string.Concat(
+               FeatureAttribute.SolutionCrawler,
+               FeatureAttribute.DiagnosticService,
+               FeatureAttribute.Classification,
+               FeatureAttribute.ReferenceHighlighting));
+
+            AssertEx.SetEqual(spans["reference"], VisualStudio.Editor.GetTagSpans(ReferenceHighlightTag.TagId));
+            AssertEx.SetEqual(spans["definition"], VisualStudio.Editor.GetTagSpans(DefinitionHighlightTag.TagId));
```
Comment (Pilchie) [inline src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReferenceHighlighting.cs]: The existing C# test verifies navigation via the next/prev reference command.  We should preserve that.

**Comment by rchande** [comment]
Test failures look unrelated to this change.

**Pull request merged**